client/components/InnerCarousel.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import InnerCarouselItem from './InnerCarouselItem.jsx';
+import './carousel.css';
+import { Animate } from 'react-rebound';
+import _ from 'underscore';
+
+
+class InnerCarousel extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentImageIndex: 0,
+    };
+    this.clickHandlerLeft = this.clickHandlerLeft.bind(this);
+    this.clickHandlerRight = this.clickHandlerRight.bind(this);
+  };
+
+  getImageOffset() {
+    const { currentImageIndex } = this.state;
+    return currentImageIndex * -335;
+  }
+
+  slideImagesLeft() {
+    console.log('slideImageLeft is firing');
+    const { currentImageIndex } = this.state;
+    const { images } = this.props.listing;
```

Comment:
In terms of separation of concerns, it might be better to have the Inner Carousel be responsible for fetching it's own images from the server, rather than inheriting them from Outer Carousel.
The API on your server is setup to respond to requests for individual listings right? As opposed to a group of listings in an Outer Carousel. So it would be a simple change if you were to implement it.